File: config/python.py
from typing import List


config_requires: List[str] = []
dev_requires: List[str] = []
install_requires: List[str] = []
build_requires: List[str] = [
    "pycmdtools",
    "pymakehelper",
    "pydmt",
    "check-jsonschema",
    "yq",
]
test_requires: List[str] = []
requires = config_requires + install_requires + build_requires + test_requires

File: config/python.lua
-- python deps for this project

dofile("config/shared.lua")

-- append every element of "src" onto "dst"
local function extend(dst, src)
    for _, value in ipairs(src) do
        table.insert(dst, value)
    end
    return dst
end

INSTALL_REQUIRES = {
    "check-jsonschema",
    "yq",
}
BUILD_REQUIRES = BUILD
TEST_REQUIRES = TEST

REQUIRES = {}
extend(REQUIRES, INSTALL_REQUIRES)
extend(REQUIRES, BUILD_REQUIRES)
extend(REQUIRES, TEST_REQUIRES)
